controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -464,18 +458,44 @@ func (r *StorageConsumerReconciler) reconcileCephRadosNamespace(
 		}
 		rns.Namespace = r.namespace
 
-		if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rns, func() error {
-			if err := controllerutil.SetControllerReference(r.storageConsumer, rns, r.Scheme); err != nil {
-				return err
+		deleteRns := false
```

Comment:
There is no point in tracking a `deleteRNS` flag. There is only a single point in the code where the delete code should run.

Reply:
sry, possible to elaborate, we need to decide b/n update or delete.